### PR TITLE
Improve the XDebug experience

### PR DESCRIPTION
**Pull request by m-vo**
Here is a followup to #4394 with some tweaks regarding running with XDebug.

Writing to `STDTERR` is less obtrusive than triggering an error [1fdb124f088de941888e2bc936399018e46e798e]. It will be colored depending on your terminal. In PHPStorm, by default, you'll basically get this in the terminal when running `vendor/bin/phpunit` / running selected tests via the run configuration:

image
image

With 1118426d3e5163cf311550f9d5b36b185134737d, running `composer run …` will default to disabling xdebug.

### CODE REVIEW

**Review by bytehead — DISMISSED** [review]
nice ❤️

File: composer.json [hunk 1]
```diff
@@ -307,14 +307,14 @@
...
-        "ecs-legacy": "@php tools/ecs/vendor/bin/ecs check *-bundle/src/Resources/contao --config tools/ecs/config/legacy.php --fix --ansi",
-        "ecs-templates": "@php tools/ecs/vendor/bin/ecs check *-bundle/src/Resources/contao/templates --config tools/ecs/config/template.php --fix --ansi",
-        "functional-tests": "@php vendor/bin/phpunit --testsuite=functional --colors=always",
-        "monorepo-tools": "@php tools/monorepo/vendor/bin/monorepo-tools composer-json --validate --ansi",
-        "phpstan": "@php tools/phpstan/vendor/bin/phpstan analyze --ansi",
-        "require-checker": "@php tools/require-checker/vendor/bin/composer-require-checker check --config-file=tools/require-checker/config.json composer.json",
-        "service-linter": "@php tools/service-linter/bin/lint-service-ids --ansi",
-        "unit-tests": "@php vendor/bin/phpunit --colors=always"
+        "ecs": "@php -dxdebug.mode=off tools/ecs/vendor/bin/ecs check *-bundle/bin *-bundle/src *-bundle/tests test-case/src tools/*/src --config tools/ecs/config/default.php --fix --ansi",
```
Comment (ausi) [inline composer.json]: Did we reevaluate the compatibility with `phpx` as noted in https://github.com/contao/contao/pull/4394#discussion_r835932957 ?

Comment (ausi) [reply composer.json]: AFAIK we concluded in a Slack huddle that `phpx` might not have worked with `@php` anyways, but did someone test it? ☺️

Comment (m-vo) [reply composer.json]: @leofeyer Could you test this with your setup? I assume no XDebug warning is output when running the `composer run unit-tests` with `phpx`. This should be the same with or without this PR.

**Comment by m-vo** [comment]
Just updated the branch, so that there are no unrelated failing tests anymore. Would be nice to get this merge anytime soon, because currently debugging is quite distressing with errors reported in between. :relaxed:

**Review by ausi — DISMISSED** [review]

File: test-case/src/WarnXdebugPhpunitExtension.php [hunk 2]
```diff
@@ -19,7 +19,7 @@ class WarnXdebugPhpunitExtension implements BeforeFirstTestHook
     public function executeBeforeFirstTest(): void
     {
         if (\is_callable('xdebug_info') && [] !== xdebug_info('mode') && ['off'] !== xdebug_info('mode')) {
-            trigger_error('XDebug is enabled, consider disabling it to speed up unit tests.', E_USER_WARNING);
+            fwrite(STDERR, "XDebug is enabled, consider disabling it to speed up unit tests.\n\n");
```
Comment (leofeyer) [inline test-case/src/WarnXdebugPhpunitExtension.php]:
```suggestion
            fwrite(STDERR, "XDebug is enabled, consider disabling it to speed up the unit tests.\n\n");
```

**Comment by leofeyer** [comment]
Why don‘t you disable Xdebug on the parent PHP process?

```
/usr/bin/php -dxdebug.mode=off /usr/local/bin/composer run unit-tests
```

**Comment by ausi** [comment]
> Why don‘t you disable Xdebug on the parent PHP process?
> 
> ```
> /usr/bin/php -dxdebug.mode=off /usr/local/bin/composer run unit-tests
> ```

This does not work because composer starts a new php process.
For the same reason we think that `phpx composer run unit-tests` does also not work for enabling xdebug.

**Comment by leofeyer** [comment]
I did some research and testing and calling the script like this does the trick for me: `XDEBUG_MODE=off composer run unit-tests` @ausi @m-vo Can you tests this please?



Found here: https://xdebug.org/docs/all_settings

**Review by aschempp — DISMISSED** [review]

**Review by leofeyer — CHANGES_REQUESTED** [review]
@m-vo Can you please revert the changes in the `composer.json` file, so I can merge this PR?

**Comment by m-vo** [comment]
> @m-vo Can you please revert the changes in the composer.json file, so I can merge this PR?

Done.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @m-vo.

**Pull request merged**